index.js

```diff
@@ -27,6 +27,28 @@ var
         var newCollection = collection.concat(records);
 
         return objectList(newCollection);
+      },
+      remove: function remove(collection, record) {
+        var newCollection = lodash.cloneDeep(collection);
+
+        lodash.remove(newCollection,
+          lodash.partial(lodash.isEqual, record));
+
+        return objectList(newCollection);
+      },
+      removeWhere: function removeWhere(collection, query) {
+        var recordsToRemove = cWhere(collection, query),
+          newCollection = lodash.reject(collection,
+            lodash.partial(lodash.find, recordsToRemove));
+
+        return objectList(newCollection);
+      },
+      removeSlice: function removeSlice(collection, start, end) {
+        var recordsToRemove = lodash.slice(collection, start, end),
+          newCollection = lodash.reject(collection,
```

Comment:
You can ignore this comment for now, though.

1. Make it work.
2. Make it right.
3. Make it fast.